packages/loader/container-loader/src/container.ts

```diff
@@ -783,8 +769,8 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
 
     public async attach(request: IRequest): Promise<void> {
         await PerformanceEvent.timedExecAsync(this.mc.logger, { eventName: "Attach" }, async () => {
-            if (this._lifecycleState !== "loaded") {
-                throw new UsageError(`containerNotValidForAttach [${this._lifecycleState}]`);
+            if (this.lifecycleStateHandler.state !== "loaded") {
```

Comment:
I think 99% of readers of this code (container.ts) will assume that lifecycleStateHandler.state === "loaded" is same as lifecycleStateHandler.isLoaded()
You would need to find different words to make it clear these are not the same things.